src/ray/raylet/agent_manager.cc

```diff
@@ -97,5 +99,81 @@ void AgentManager::StartAgent() {
   monitor_thread.detach();
 }
 
+void AgentManager::CreateRuntimeEnv(const std::string &serialized_runtime_env,
+                                    CreateRuntimeEnvCallback callback) {
+  if (runtime_env_agent_client_ == nullptr) {
+    RAY_LOG(INFO) << "No available agent connection. Retry to create runtime env "
+                  << serialized_runtime_env << " later.";
+    delay_executor_([this, serialized_runtime_env,
+                     callback] { CreateRuntimeEnv(serialized_runtime_env, callback); },
+                    RayConfig::instance().agent_retry_interval_ms());
+    return;
+  }
+  rpc::CreateRuntimeEnvRequest request;
+  request.set_serialized_runtime_env(serialized_runtime_env);
+  runtime_env_agent_client_->CreateRuntimeEnv(
+      request, [this, serialized_runtime_env, callback](
+                   Status status, const rpc::CreateRuntimeEnvReply &reply) {
+        if (status.ok()) {
+          if (reply.status() == rpc::AGENT_RPC_STATUS_OK) {
+            callback(true);
+          } else {
+            RAY_LOG(ERROR) << "Failed to create runtime env: " << serialized_runtime_env
+                           << ", error message: " << reply.error_message();
+            callback(false);
+          }
+
+        } else {
+          RAY_LOG(ERROR) << "Failed to create the runtime env: " << serialized_runtime_env
+                         << ", status = " << status
+                         << ", maybe there are some network problems, retry it later.";
+          runtime_env_agent_client_.reset();
```

Comment:
why reset the connection? Will the agent re-register to the raylet?

Reply:
Sorry, I have checked agent code just now. No re-register now in agent. We should delete this "reset" and reserve this grpc client.